tools/readlink.py

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python
+"""Canonicalizes a path and follows any symlinks.
+
+This is the equivalent of `readlink -f` on many Linux systems. This is
+useful as there are often differences in readlink on different
+platforms.
+
+"""
+from __future__ import print_function
+import os
+import sys
+
+print(os.path.realpath(sys.argv[1]))
```

Comment:
Does this not just call `readlink -f` internally? Are we guaranteed a consistent result if we use the python version?